makefiles/boot/riotboot.mk

```diff
@@ -9,26 +9,29 @@ CFLAGS += -I$(BINDIR)/riotbuild
 
 HEADER_TOOL_DIR = $(RIOTBASE)/dist/tools/riotboot_gen_hdr
 HEADER_TOOL ?= $(HEADER_TOOL_DIR)/bin/genhdr
-BINDIR_APP = $(BINDIR)/$(APPLICATION)
+BINDIR_RIOTBOOT = $(BINDIR)/riotboot_files
+
+$(BINDIR_RIOTBOOT): $(CLEAN)
+	$(Q)mkdir -p $(BINDIR_RIOTBOOT)
 
 #
 export SLOT0_OFFSET SLOT0_LEN SLOT1_OFFSET SLOT1_LEN
 
 # Mandatory APP_VER, set to epoch by default
-EPOCH := $(shell date +%s)
+EPOCH = $(call memoized,EPOCH,$(shell date +%s))
```

Comment:
What's the reasoning behind that change?

Reply:
Avoiding a shell call if not used

Reply:
I can split

Reply:
It's fine, I was just not familiar with `call memoized`